File: C++/calcolatrice/calcolatrice.cpp
//Autore: Gianluca Soban
//Data: 06.10.23

//Calcolatrice

#include <iostream>
#include <cstdlib>
#include <math.h>
using namespace std;

int main()
{
    int valore = 0;
    float num1, num2, risultato;
    int mod1, mod2; //numeri che servono per il modulo

    cout<<"Premere 1 per fare una somma.";
    cout<<"\nPremere 2 per fare una sottrazione.";
    cout<<"\nPremere 3 per fare una moltiplicazione.";
    cout<<"\nPremere 4 per fare una divisione.";
    cout<<"\nPremere 5 per fare il modulo.";
    cout<<"\nPremere 6 per fare la radice quadrata." <<endl;
    cin>>valore;

    switch (valore)
    {
        case 1:
            cout<<"Inserire il primo numero per la somma: ";
            cin>>num1;
            cout<<"\nInserire il secondo numero per la somma: ";
            cin>>num2;

            risultato = num1+num2;
            cout<<"\nIl risultato e': "<<risultato;
            break;

        case 2:
            cout<<"Inserire il primo numero per la sottrazione: ";
            cin>>num1;
            cout<<"Inserire il secondo numero per la sottrazione: ";
            cin>>num2;

            risultato = num1-num2;
            cout<<"\nIl risultato e': "<<risultato;
            break;

        case 3:
            cout<<"Inserire il primo numero per la moltiplicazione: ";
            cin>>num1;
            cout<<"Inserire il secondo numero per la moltiplicazione: ";
            cin>>num2;

            risultato = num1*num2;
            cout<<"\nIl risultato e': "<<risultato;
            break;

        case 4:
            cout<<"Inserire il numero da dividere: ";
            cin>>num1;
            cout<<"Inserire il divisore: ";
            cin>>num2;

            if(num2>0)
            {
                risultato = num1/num2;
                cout<<"\nIl risultato e': "<<risultato;
            }
            else
                cout<<"Impossibilie dividere per 0.";
            break;

        case 5:
            cout<<"Inserire il numero intero da dividere: ";
            cin>>mod1;
            cout<<"Inserire il divisore intero: ";
            cin>>mod2;

            risultato = mod1/mod2;
            cout<<"\nIl risultato e': "<<risultato;
            cout<<"\nIl resto della divisione e': "<<mod1%mod2;
            break;

        case 6:
            cout<<"Inserire il numero per fare la radice quadrata: ";
            cin>>num1;

            if(num1>0)
            {
                risultato = sqrt(num1);
                cout<<"La radice quadrata di "<<num1<<" e': "<<risultato;
            }
            else
                cout<<"Impossibiole fare la radice.";
            break;

        default:
            cout<<"Valore non supportato.";
            break;
    }
return 0;
}

File: Java/Calcolatrice.java
import java.util.Scanner;

public class Calcolatrice{

    static Scanner tastiera = new Scanner(System.in);

    public static float Somma(){
        float num1=0, num2=0, risultato=0;

        System.out.print("Inserisci il primo numero per la somma: ");
        num1 = tastiera.nextFloat();
        System.out.print("Inserisci il secondo numero per la somma: ");
        num2 = tastiera.nextFloat();

        risultato = num1+num2;

        return risultato;
    }

    public static float Sottrazione(){
        float num1=0, num2=0, risultato=0;

        System.out.print("Inserisci il primo numero per la sottrazione: ");
        num1 = tastiera.nextFloat();
        System.out.print("Inserisci il secondo numero per la sottrazione: ");
        num2 = tastiera.nextFloat();

        risultato = num1-num2;

        return risultato;
    }

    public static float Moltiplicazione(){
        float num1=0, num2=0, risultato=0;

        System.out.print("Inserisci il primo numero per la moltiplicazione: ");
        num1 = tastiera.nextFloat();
        System.out.print("Inserisci il secondo numero per la moltiplicazione: ");
        num2 = tastiera.nextFloat();

        risultato = num1*num2;

        return risultato;
    }

    public static float Divisione(){
        float num1=0, num2=0, risultato=0;

        System.out.print("Inserisci il primo numero per la divisione: ");
        num1 = tastiera.nextFloat();
        System.out.print("Inserisci il secondo numero per la divisione: ");
        do{
            num2 = tastiera.nextFloat();
            if(num2 == 0)
                System.out.print("Impossibile dividere per 0, inserire di niovo: ");
        }while(num2 == 0);

        risultato = num1/num2;

        return risultato;
    }

    public static void Calcolatore(int scelta){
        float risultato=0;

        switch (scelta) {
            case 1:
                risultato = Somma();
                System.out.print("La somma e' pari a: " + risultato + ".");
                break;

            case 2:
                risultato = Sottrazione();
                System.out.print("La sottrazione e' pari a: " + risultato + ".");
                break;

            case 3:
                risultato = Moltiplicazione();
                System.out.print("La moltiplicazione e' pari a: " + risultato + ".");
                break;

            case 4:
                risultato = Divisione();
                System.out.print("La divisione e' pari a: " + risultato + ".");
                break;
        
            default:
                System.err.print("Errore nell'esecuzione.");
                break;
        }
    }
    public static void main(String[] args){
        int scelta;

        System.out.println("Premere 1 per fare una somma.");
        System.out.println("Premere 2 per fare una sottrazione.");
        System.out.println("Premere 3 per fare una moltiplicazione.");
        System.out.println("Premere 4 per fare una divisione.");
        System.out.print("Inserisci la tua scelta: ");
        do{
            scelta = tastiera.nextInt();
            if(scelta<1 || scelta > 4)
                System.err.print("Errore, inserisci di nuovo: ");
        }while(scelta<1 || scelta > 4);
        
        Calcolatore(scelta);
    } 
}